app/Project.php

```diff
@@ -145,4 +145,84 @@ public function user()
     {
         return $this->belongsTo(User::class);
     }
+
+    /**
+     *
+     *
+     * @return 
+     */
+    public function followers()
+    {
+        return $this->hasMany(ProjectFollower::class);
+    }
+
+    /**
+     * Get the count of followers for the project
+     *
+     * @return
+     */
+    public function getFollowerCount()
+    {
+        return count(ProjectFollower::where('project_id', '=', $this->id)->get());
+    }
+
+    /**
+     * Check if user is a follower of this project
+     *
+     * @return
+     */
+    public function isUserFollower()
+    {
+        if (!Auth::guest()) {
+            $isUserFollower = ProjectFollower::where('user_id', '=', Auth::user()->id)->where('project_id', '=', $this->id)->first();
+            if ($isUserFollower != null)
+                $isUserFollower = true;
+            else
+                $isUserFollower = false;
+        }
+        else
+            $isUserFollower = false;
+
+        return $isUserFollower;
+    }
+
+    /**
+     * Adds a follower to the project
+     *
+     * @return
+     */
+    public function addFollower()
```

Comment:
This one too, but here there's no check for auth, presumably because the user won't be authenticated. When i try to use a direct link in incognito i get the 403 page, so that's good.